configure.ac

```diff
@@ -1,313 +1,127 @@
-#                                               -*- Autoconf -*-
-# Process this file with autoconf to produce a configure script.
+dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.59)
+
 AC_INIT(Normaliz, 3.7.4)
+
 AC_SUBST(LIBNORMALIZ_VERSION_MAJOR, 3)
 AC_SUBST(LIBNORMALIZ_VERSION_MINOR, 7)
 AC_SUBST(LIBNORMALIZ_VERSION_PATCH, 4)
 AC_SUBST(LIBNORMALIZ_VERSION_STRING, "$PACKAGE_VERSION")
-AM_INIT_AUTOMAKE(foreign)
-AC_CONFIG_FILES([Makefile
-	source/Makefile
-	source/libnormaliz/version.h
-	example/Makefile
-	test/Makefile
-	source/libnormaliz/nmz_config.h])
+
 AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_AUX_DIR([.])
+AM_INIT_AUTOMAKE([1.14.1])
+
+AC_CANONICAL_HOST
+
+dnl Find C++ Compiler
 AC_PROG_CXX
+
+dnl Set INSTALL or use install-sh.
+AC_PROG_INSTALL
+
+dnl We build our library with libtool.
+LT_INIT
+
+dnl We want to use some C++14 features
 AX_CXX_COMPILE_STDCXX(14, , optional)
+
 AS_IF([test x$HAVE_CXX14 = x0],
+  dnl We require C++0x features
   [ AX_CXX_COMPILE_STDCXX(0x, , mandatory) ])
 
-AC_PROG_LIBTOOL
+dnl Check for required/optional libraries
 AC_LANG(C++)
 
-## Test for GMP
-AC_ARG_WITH(gmp,
-   AS_HELP_STRING([--with-gmp=DIR],
-                  [Use the GMP library installed in installation prefix DIR.]),
-   [ LDFLAGS="-L$with_gmp/lib $LDFLAGS"
-     CPPFLAGS="-I $with_gmp/include $CPPFLAGS"
-   ])
-AC_MSG_CHECKING([whether GMP C++ library is available])
-LIBS_SAVED="$LIBS"
-LIBS="$LIBS -lgmpxx -lgmp"
-AC_LINK_IFELSE(
-   [AC_LANG_PROGRAM([[#include <gmpxx.h>
-                     ]],
-                    [[mpz_class a;
-                     ]])],
-   [AC_MSG_RESULT([yes])],
-   [AC_MSG_ERROR([GMP C++ library not found. Make sure it was compiled with --enable-cxx])])
-LIBS="$LIBS_SAVED"
+AC_CHECK_HEADERS([boost/dynamic_bitset.hpp], , AC_MSG_ERROR([boost headers not found]))
 
-## Test for OpenMP parallelization.
-AC_ARG_ENABLE([openmp],
-    [AS_HELP_STRING([--enable-openmp@<:@=ARG@:>@],
-      [enable parallelization with OpenMP @<:@default=check@:>@])],
-    [:],
-    [enable_openmp=check])
-AS_IF([test "x$enable_openmp" != xno],
-  [AX_OPENMP([have_openmp=yes],
-	     [have_openmp=no])])
+# We use GMP's C++ wrapper but we also interface with GMP's C interface directly.
+AC_CHECK_LIB([gmp], [__gmpz_init], [have_gmp=yes], AC_MSG_ERROR([GMP library not found]))
```

Comment:
So `--with-gmp` has been removed without a replacement? Is that intentional? Can you explain the motivation?

Also, you then need to adjust places using it, i.e. at least `install_normaliz.sh`

Reply:
I removed it because I don't see why it should be used to specify a search path. That's what LDFLAGS is good for. As for `install_normaliz.sh`, sure that needs to be adapted, thanks for pointing that out. (Feel free to patch it up.) I don't really understand what that script is good for though.

Reply:
Fair enough. So, if everybody is happy with dropping all the optional prefixes from the `--with` arguments (and adapting code relying on that, and enhance the build instructions to explain how to do it instead, etc.), I personally don't mind -- I have no troubles building Normaliz either way. But in my experience (i.e., maintenance of other software using configure scripts, with similar options, and the support feedback we get from users) , a lot of people expect that they can do e.g. `--with-gmp=PATH`, and if this works, they are confused, and ask us for help (and it doesn't matter if we explain how to do it somewhere --- few people read those instructions, sadly. Well, I usually don't anymore, too ;-).

So, it's a tradeoff: simplicity versus user-friendliness. Is the extra simplicity worth the fact that it may be harder for some people to compile Normaliz? (Though of course this PR may simplify other things, so the overall balance probably needs to be taken into account, too)